pkg/typesystem/connected_types_test.go

```diff
@@ -0,0 +1,165 @@
+package typesystem
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/openfga/openfga/pkg/testutils"
+)
+
+func TestTypesystemAreTypesConnectedViaRelation(t *testing.T) {
+	model := `
+		model
+			schema 1.1
+		type user
+		type employee
+
+		type document
+			relations
+				define computed: computed2
+				define computed2: direct_user
+				define direct_user: [user]
+				define direct_employee: [employee]
+				define direct_user_or_computed: [user] or direct_user
+				define direct_user_and_employee: [ user, employee ]
+				define union_user_or_employee: direct_employee or direct_user
+				define intersection_user_and_employee: direct_user and direct_employee
+				define intersection_direct_and_computed_user: [user] and direct_user
+	`
+
+	tests := []struct {
+		expectedAreConnected      bool
+		expectedTerminalRelations []string
+		objectType                string
+		relation                  string
+		subjectType               string
+	}{
+		// // Intersection
+		{
+			objectType:                "document",
+			relation:                  "intersection_user_and_employee",
+			subjectType:               "employee",
+			expectedAreConnected:      false,
```

Comment:
This is WIP, right? The expectation here should be true because `employee` is reachable from `document`.

Reply:
First, please read PR description.

Secondly, `employee` is effectively not reachable because there are no types that intersect between the `intersection_user_and_employee` relation.

Reply:
I did read the PR description, I understood that the code was WIP but i didn't get whether the test was also WIP